sklearn/utils/estimator_checks.py

```diff
@@ -373,7 +378,23 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
-def parametrize_with_checks(estimators):
+def _get_xfail_checks(estimator, strict_mode):
+    # Return the checks that are in the estimator's _xfail_checks tag, along
+    # with the strict checks if strict_mode is False.
+    xfail_checks = estimator._get_tags()['_xfail_checks'] or {}
+
+    if not strict_mode:
+        strict_checks = {
+            _set_check_estimator_ids(check):
```

Comment:
This one is a bit unfortunate. I mean we already have the raw function objects, but I guess not re-using this mechanism means we have to handle partials functions. So it's probably OK for now.

Reply:
Alternatively we can also directly use `_set_check_estimator_ids(check)` in the `_STRICT_CHECKS` dict? that would be a dict of strings instead of a dict of functions

I'm only using the same logic as for the `_xfail_checks` tag (which I find... a bit sloppy, I'll give you  that ;) )

Reply:
> Alternatively we can also directly use _set_check_estimator_ids(check) in the _STRICT_CHECKS dict?

No strong feeling about it. It's more a side comment, I'm OK with the proposed solution as well.